pywbem_mock/_defaultinstanceprovider.py

```diff
@@ -0,0 +1,527 @@
+              Must not be None.
+            * `host`: value ignored.
+
+          LocalOnly (:class:`py:bool`):
+            This parameter is used to control the exclusion of inherited
+            properties in the returned instances. :term:`DSP0200` version 1.2
+            has deprecated this parameter and allows servers to honor it, or to
+            treat it as False, as long as that is done consistently.
+
+            This provider ignores the parameter and treats it as False, so that
+            inherited properties are included in the returned instances. This
+            is consistent with the recommendation in DSP0200 for clients to set
+            the parameter to False.
+
+          IncludeQualifiers (:class:`py:bool`):
+            Indicates that qualifiers are to be included in the returned
+            instance, as follows:
+
+            * If `False`, qualifiers are not included.
+            * If `True`, qualifiers are included if the WBEM server implements
+              support for this parameter.
+            * If `None`, the :term:`DSP0200` defined server default `False`
+              is used.
+
+            This parameter has been deprecated in :term:`DSP0200`. Clients
+            cannot rely on qualifiers to be returned in this operation.
+
+          IncludeClassOrigin (:class:`py:bool`):
+            Indicates that class origin information is to be included on each
+            property in the returned instance, as follows:
+
+            * If `False`, class origin information is not included.
+            * If `True`, class origin information is included.
+            * If `None`, uses :term:`DSP0200` server default `False`.
+
+          PropertyList (:term:`string` or :term:`py:iterable` of :term:`string`):
+            An iterable specifying the names of the properties (or a string
+            that defines a single property) to be included in the returned
+            instance (case independent).
+
+            An empty iterable indicates to include no properties.
+
+            If `None`, all properties are included.
+
+        Returns:
+
+            A :class:`~pywbem.CIMInstance` object that is a representation of
+            the retrieved instance.
+            Its `path` attribute is a :class:`~pywbem.CIMInstanceName` object
+            with its attributes set as follows:
+
+            * `classname`: Name of the creation class of the instance.
+            * `keybindings`: Keybindings of the instance.
+            * `namespace`: Name of the CIM namespace containing the instance.
+            * `host`: `None`, indicating the WBEM server is unspecified.
+
+        Raises:
+
+            CIMError:  CIM_ERR_INVALID_NAMESPACE, CIM_ERR_INVALID_PARAMETER
+              CIM_ERR_NOT_FOUND
+        """  # noqa: E501
+        # pylint: enable=line-too-long
+
+        namespace = InstanceName.namespace
+        self.validate_namespace(namespace)
```

Comment:
This check could be done before calling the provider.

Reply:
DONE